### PR TITLE
Fix compiler warnings (-Wlogical-op)

### PR DESCRIPTION
**Pull request by fhuberts**
See #784

### CODE REVIEW

File: src/oatpp/network/tcp/Connection.cpp [hunk 1]
```diff
@@ -115,7 +115,19 @@ v_io_size Connection::write(const void *buff, v_buff_size count, async::Action&
 
   if(result < 0) {
     auto e = errno;
-    if(e == EAGAIN || e == EWOULDBLOCK){
+
```
Comment (lganzzzo) [inline src/oatpp/network/tcp/Connection.cpp:118]: Let's suppress warning for the whole function - 
just to make code more readable.

```cpp
#ifdef __GNUC__
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wlogical-op"
#endif

function {
 ...
}

#ifdef __GNUC__
#pragma GCC diagnostic pop
#endif

```

File: src/oatpp/network/tcp/Connection.cpp [hunk 2]
```diff
@@ -173,7 +185,19 @@ v_io_size Connection::read(void *buff, v_buff_size count, async::Action& action)
...
```
Comment (lganzzzo) [inline src/oatpp/network/tcp/Connection.cpp:188]: Let's suppress warning for the whole function - 
just to make code more readable.

```cpp
#ifdef __GNUC__
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wlogical-op"
#endif

function {
 ...
}

#ifdef __GNUC__
#pragma GCC diagnostic pop
#endif

```

**Review by lganzzzo — CHANGES_REQUESTED** [review]

File: src/oatpp/network/tcp/Connection.cpp (hunk 1 above)
Comment (fhuberts) [reply src/oatpp/network/tcp/Connection.cpp:123]: done

File: src/oatpp/network/tcp/Connection.cpp (hunk 2 above)
Comment (fhuberts) [reply src/oatpp/network/tcp/Connection.cpp:193]: done

**Review by lganzzzo — APPROVED** [review]

**Comment by lganzzzo** [comment]
merged

**Pull request merged**